### PR TITLE
CATROID-Kotlin_static_analysis add kotlin static analysis and style c…

### PR DESCRIPTION
**Pull request by LPeteR90**
…hecks

Further
- gradle: fix warning regarding "configuration compile is deprecated"
- gradle: now support Java 8 language features

### CODE REVIEW

**Comment by LPeteR90** [comment]
Some rules in the detekt configuration are subject to discussion and change. Code style is basically the official Kotlin style. 

Feel free to contribute any proposals.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Wed Mar 08 18:16:17 CET 2017
+#Thu Mar 21 20:32:52 CET 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
```
Comment (ThomasSchwengler) [inline gradle/wrapper/gradle-wrapper.properties]: I think there was some issue (in the past?) preventing paintroid from upgrading to gradle > 4.4 - I think it was some build dependency. I will see if I find out what issues we had. Might just have been our bintray upload plugin, which won't be an issue for catroid.

File: catroid/build.gradle [hunk 2]
```diff
@@ -24,14 +24,19 @@
 buildscript {
     repositories {
         maven { url 'https://maven.fabric.io/public' }
+        google()
         jcenter()
     }
     dependencies {
         classpath 'io.fabric.tools:gradle:1.21.5'
-        classpath 'com.google.gms:google-services:3.0.0'
+        classpath 'com.google.gms:google-services:4.2.0'
```
Comment (ThomasSchwengler) [inline catroid/build.gradle]: So glad that this is finally being updated :+1:

Comment (ThomasSchwengler) [reply catroid/build.gradle]: How likely is it that this has runtime side effects?

**Comment by mfuchs** [comment]
The Jenkinsfile changes look ok to me.

**Comment by wslany** [comment]
Super!

File: catroid/build.gradle [hunk 3]
```diff
@@ -290,6 +303,10 @@ android {
     }
 
     flavorDimensions "default"
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
```
Comment (rschluesselbauer) [inline catroid/build.gradle]: 🥇

**Pull request merged**